### PR TITLE
Cake Addin for getting Nerdbank.GitVersioning VersionOracle.

### LINKED ISSUE #148: Cake Build Addin Version
Hi there,
I have noticed that your Tool is executed Via a Powershell script, which triggers various DLL Libraries.
Have you considering adding Cake Build Addin Support?

[Cake](https://cakebuild.net/) is a tool for automating building projects using C# and Roslyn (It stands for C# Make), it also accepts addins and tools via NuGet import, if your .dlls support .NET Core, it might be trivial to create a Cake Alias for it.
This way you can avoid Powershell, and allow for simple cross platform support.

### PR DESCRIPTION
**Pull request by clcrutch**
## Cake Addin for getting Nerdbank.GitVersioning VersionOracle.
This is a cake addin for getting the VersionOracle for the current git repo.

Fixes #148

### CODE REVIEW

**Comment by clcrutch** [comment]
I'm currently working on addressing these build failures. Will report back.

**Comment by clcrutch** [comment]
Sorry for the delay in getting the build issues addressed.

File: src/Cake.GitVersioning/Cake.GitVersioning.csproj [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Authors>Chris Crutchfield, Andrew Arnott</Authors>
+    <Company>andarno</Company>
+    <Description>Cake wrapper for Nerdbank.GitVersioning.  Stamps your assemblies with semver 2.0 compliant git commit specific version information and provides NuGet versioning information as well.</Description>
+    <Copyright>Copyright © Andrew Arnott</Copyright>
+    <PackageTags>git commit versioning version assemblyinfo</PackageTags>
+    <PackageLicenseUrl>https://raw.githubusercontent.com/AArnott/Nerdbank.GitVersioning/39ac76891c/LICENSE.txt</PackageLicenseUrl>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <PackageProjectUrl>http://github.com/aarnott/Nerdbank.GitVersioning</PackageProjectUrl>
+    <SignAssembly>false</SignAssembly>
```
[the remark this answers is not in the document]
Comment (AArnott) [reply src/Cake.GitVersioning/Cake.GitVersioning.csproj]: I see... Cake.Core itself isn't strong name signed.

[the remark this answers is not in the document]
Comment (clcrutch) [reply src/Cake.GitVersioning/Cake.GitVersioning.csproj]: Yeah, I initially attempted to sign the assembly, but also discovered that Cake.Core was not signed.  I would be willing to submit an issue to the Cake team to see if they'd be willing to strong name sign Cake.Core.  I am unsure if that would affect anything down stream of them though.

[the remark this answers is not in the document]
Comment (AArnott) [reply src/Cake.GitVersioning/Cake.GitVersioning.csproj]: It probably would impact their existing users. And my guess for a package that popular is that if they're not signed, they probably are opposed to the idea. It's OK. We can fall in line for the cake plugin.

**Comment by AArnott** [comment]
@clcrutch Can you confirm that the package we're building after my changes to your PR works?

**Comment by AArnott** [comment]
Also, can you update the docs in this repo to include something about this feature?

**Comment by clcrutch** [comment]
@AArnott I just confirmed that the changes still work after your changes.  I'll update the docs to include information about this feature and let you know when I am finished.

**Comment by AArnott** [comment]
@clcrutch Any update on the docs? I've fixed #185 so I'm about ready to accept this PR after your doc addition.

**Comment by clcrutch** [comment]
I've added docs.  There are a few mistakes I'm working on correcting.

**Comment by clcrutch** [comment]
I've updated the docs.  I think I'm content with them.  I'd love some feedback though.

**Comment by clcrutch** [comment]
I'm not sure why the build failed.  The build failure appears to happen after it has already built my code.  It also builds fine on my end.
image

**Comment by clcrutch** [comment]
All builds now pass.

**Comment by clcrutch** [comment]
Sorry about the noise.  Was reviewing the pull request and accidentally closed the PR.

Otherwise, I think I'm content with the changes here.  Is there anything else that I should do on my end?

**Comment by AArnott** [comment]
Great. I just got back from a vacation where I had no Internet access, so I'm catching up. I look forward to reviewing this again and merging.

**Comment by AArnott** [comment]
Why are all these markdown files that describe the public API checked into source? If they were generated, they need to be generated during the build--not as a once off plus checkin. Otherwise it becomes a maintenance nightmare. Can that be done?

**Comment by AArnott** [comment]
(just to be clear, when I asked for documentation I was only asking for the `doc/cake.md` file that you added.)

File: src/Nerdbank.GitVersioning.sln [hunk 2]
```diff
@@ -26,7 +26,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Nerdbank.GitVersioning.Task
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "MSBuildExtensionTask", "MSBuildExtensionTask\MSBuildExtensionTask.csproj", "{568CD02D-49C4-49DB-A34D-2DE9D7A0FE85}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "nbgv", "nbgv\nbgv.csproj", "{EF4DAF23-6CE9-48C5-84C5-80AC80D3D07D}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Cake.GitVersioning", "Cake.GitVersioning\Cake.GitVersioning.csproj", "{1F267A97-DFE3-4166-83B1-9D236B7A09BD}"
 EndProject
```
Comment (AArnott) [inline src/Nerdbank.GitVersioning.sln]: It looks like you resolved a merge conflict by blowing away the competing addition. I need nbgv to stick around. :)

Comment (AArnott) [reply src/Nerdbank.GitVersioning.sln]: I fixed this.

---
In reply to: [211299223](https://github.com/AArnott/Nerdbank.GitVersioning/pull/171#discussion_r211299223) [](ancestors = 211299223)

File: src/Nerdbank.GitVersioning.sln.orig [hunk 3]
```diff
@@ -0,0 +1,84 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
[the remark this answers is not in the document]
Comment (AArnott) [reply src/Nerdbank.GitVersioning.sln.orig]: I fixed this.

---
In reply to: [211302974](https://github.com/AArnott/Nerdbank.GitVersioning/pull/171#discussion_r211302974) [](ancestors = 211302974)

**Pull request merged**